lua/lazyvim/plugins/extras/lang/rust.lua

```diff
@@ -45,82 +45,60 @@ return {
-          liblldb_path = extension_path .. "lldb\\bin\\liblldb.dll"
-        elseif vim.fn.has("mac") == 1 then
-          liblldb_path = extension_path .. "lldb/lib/liblldb.dylib"
-        else
-          liblldb_path = extension_path .. "lldb/lib/liblldb.so"
-        end
-        adapter = require("rust-tools.dap").get_codelldb_adapter(codelldb_path, liblldb_path)
-      end
-      return {
-        dap = {
-          adapter = adapter,
+    "mrcjkb/rustaceanvim",
+    version = "^3", -- Recommended
+    ft = { "rust" },
+    opts = {
+      server = {
+        settings = {
+          -- rust-analyzer language server configuration
+          ["rust-analyzer"] = {
+            cargo = {
+              allFeatures = true,
+              loadOutDirsFromCheck = true,
+              runBuildScripts = true,
+            },
+            -- Add clippy lints for Rust.
+            checkOnSave = {
+              allFeatures = true,
+              command = "clippy",
+              extraArgs = { "--no-deps" },
+            },
+            procMacro = {
+              enable = true,
+              ignored = {
+                ["async-trait"] = { "async_trait" },
+                ["napi-derive"] = { "napi" },
+                ["async-recursion"] = { "async_recursion" },
+              },
+            },
+          },
         },
-        tools = {
-          on_initialized = function()
-            vim.cmd([[
-                  augroup RustLSP
-                    autocmd CursorHold                      *.rs silent! lua vim.lsp.buf.document_highlight()
-                    autocmd CursorMoved,InsertEnter         *.rs silent! lua vim.lsp.buf.clear_references()
-                    autocmd BufEnter,CursorHold,InsertLeave *.rs silent! lua vim.lsp.codelens.refresh()
-                  augroup END
-                ]])
+      }
+    },
+    config = function(_, opts)
+      vim.g.rustaceanvim = vim.tbl_deep_extend("force", {
+        -- LSP configuration
+        server = {
+          on_attach = function(client, bufnr)
+            -- register which-key mappings
+            local wk = require("which-key")
+            wk.register({
+              ["<leader>cR"] = { function() vim.cmd.RustLsp("codeAction") end, "Code Action" },
+              ["<leader>dr"] = { function() vim.cmd.RustLsp("debuggables") end, "Rust debuggables" },
+            }, { mode = "n", buffer = bufnr })
           end,
         },
-      }
-    end,
-    config = function() end,
+      }, opts or {})
+    end
   },
 
   -- Correctly setup lspconfig for Rust 🚀
   {
     "neovim/nvim-lspconfig",
     opts = {
       servers = {
-        -- Ensure mason installs the server
-        rust_analyzer = {
-          keys = {
-            { "K", "<cmd>RustHoverActions<cr>", desc = "Hover Actions (Rust)" },
-            { "<leader>cR", "<cmd>RustCodeAction<cr>", desc = "Code Action (Rust)" },
-            { "<leader>dr", "<cmd>RustDebuggables<cr>", desc = "Run Debuggables (Rust)" },
-          },
```

Comment:
I hope you don't take it as me bothering you, but maybe it would be better for the end user to expose the keys of this PR in `lspconfig` instead of `rustaceanvim`'s `on_attach` function, so it's easier for them to change them if they want to? Otherwise the user would have to mess with the `config` function to do that?

Reply:
I'm not very familiar with how lazy works, but rustaceanvim is opinionated in the sense that it doesn't want `lspconfig` to handle anything related to `rust_analyzer`. 

Would disabling everything but the keys in the `lspconfig` work?

Reply:
I tried it and didn't work, that's why the final commit put them in the `on_attach` function instead.